### ISSUE TITLE
Turbo Pack only last group layout considered for page file

### ISSUE THREAD

**Issue by SacredAI** [vercel/next.js#57551]
### Link to the code that reproduces this issue

https://github.com/SacredAI/nextjs-group-layout-minimal-repo

### To Reproduce

1. Start the application with turbopack (next dev --turbo)
2. Attempt to navigate to index page
3. Should result in a 404 despite index page being in app/(anothergroup)/page.tsx
4. Rename (anothergroup) to (z)
5. Try again and the page will appear

### Current vs. Expected behavior

Currently it appears like turbopack only checks the alphabetically last grouplayout for a page component to render.
I expect to not need to name my group layouts in a certain way in order for the page navigation to be unaffected while using turbo pack.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Home
Binaries:
  Node: 18.17.0
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.0.1-canary.0
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.2.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Turbopack (--turbo)

**Comment by SacredAI** [comment]
I will also note that I believe this only occurs when a layout file is present within a preceding group layout

**Comment by ForsakenHarmony** [comment]
@SacredAI this should be fixed with #57647
